lib/tlscontext.h

```diff
@@ -64,8 +65,21 @@ typedef struct _TLSSession
   TLSSessionVerifyFunc verify_func;
   gpointer verify_data;
   GDestroyNotify verify_data_destroy;
+  TLSPeerInformation *peer_info;
 } TLSSession;
 
+#define X509_MAX_CN_LEN 64
+#define X509_MAX_O_LEN 64
+#define X509_MAX_OU_LEN 64
+
+typedef struct _TLSPeerInformation
+{
+  int found;
+  gchar o[X509_MAX_O_LEN];
+  gchar ou[X509_MAX_OU_LEN];
+  gchar cn[X509_MAX_CN_LEN];
+} TLSPeerInformation;
+
```

Comment:
I am not sure 64 characters would be enough. Is there a specified/recommended limit somewhere?